contracts/gas-estimation/InterchainGasEstimation.sol

```diff
@@ -166,8 +162,49 @@ abstract contract InterchainGasEstimation is IInterchainGasEstimation {
         uint256 units = (txDataNonZeroGasEIP2028 * l1Bytes) / 2;
 
         return
-            (gasInfo.relativeGasPrice * (units + estimationPaddingUnits) * (oneInBips + estimationPaddingBasisPoints)) /
-            oneInBips;
+            (l1GasInfo.relativeGasPrice *
+                (units + estimationPaddingUnits) *
+                (oneInBips + estimationPaddingBasisPoints)) / oneInBips;
+    }
+
+    /**
+     * @notice Computes the L1 to L2 fee for a contract call on the Scroll chain.
+     * @param payload The payload of the contract call
+     * @param l1GasInfo The L1 gas info
+     * @return l1DataFee The L1 to L2 data fee
+     */
+    function scrollL1Fee(bytes calldata payload, GasInfo storage l1GasInfo) internal view returns (uint256 l1DataFee) {
+        // https://docs.scroll.io/en/developers/guides/estimating-gas-and-tx-fees/
+        // Reference https://github.com/scroll-tech/scroll/blob/af2913903b181f3492af1c62b4da4c1c99cc552d/contracts/src/L2/predeploys/L1GasPriceOracle.sol#L63-L86
+        uint256 overhead = 2500;
+        uint256 l1BaseFee = 15_602_435_409;
```

Comment:
how can we hardcode `l1BaseFee`? `l1GasInfo.relativeGasPrice` is the L1 gas price (base fee + priority fee) that we are already multiplying

Reply:
good catch